### PR TITLE
Update utils.ts

### PR DESCRIPTION
**Pull request by doggy8088**
Performance tuning for RegEx and while loop

### CODE REVIEW

**Review by kyomind — APPROVED** [review]
I am not familiar with TypeScript, only reviewed the RegEx pattern.😷

**Comment by doggy8088** [comment]
It's okay. I'm just give you some suggestions. You can close my PR and modify part of the code as you want. 😄

**Comment by Talljack** [comment]
> It's okay. I'm just give you some suggestions. You can close my PR and modify part of the code as you want. 😄

yeah, I know and thank you very much.

File: src/utils.ts [hunk 1]
```diff
@@ -2,17 +2,14 @@ import type { TextEditorEdit } from 'vscode'
...
-    const chinese = match[1]
-    const english = match[2]
-    line = line.replace(chinese + english, `${chinese} ${english}`)
-    changed = true
-    match = regexCE.exec(line) || regexEC.exec(line)
-  }
+  do {
+    line = line.replace(regexCE, (_, chinese, english) => `${chinese} ${english}`)
+    line = line.replace(regexEC, (_, english, chinese) => `${english} ${chinese}`)
+    changed = true;
```
Comment (Talljack) [inline src/utils.ts:11]:
```suggestion
    changed = true
```

File: src/utils.ts [hunk 2]
```diff
@@ -2,17 +2,14 @@ import type { TextEditorEdit } from 'vscode'
...
+  } while (regexCE.test(line) || regexEC.test(line));
```
Comment (Talljack) [inline src/utils.ts:12]:
```suggestion
  } while (regexCE.test(line) || regexEC.test(line))
```

File: src/utils.ts [hunk 3]
```diff
@@ -2,17 +2,14 @@ import type { TextEditorEdit } from 'vscode'
...
-    const chinese = match[1]
-    const english = match[2]
-    line = line.replace(chinese + english, `${chinese} ${english}`)
+  do {
+    line = line.replace(regexCE, (_, chinese, english) => `${chinese} ${english}`)
+    line = line.replace(regexEC, (_, english, chinese) => `${english} ${chinese}`)
     changed = true
-    match = regexCE.exec(line) || regexEC.exec(line)
-  }
+  } while (regexCE.test(line) || regexEC.test(line))
```
Comment (Talljack) [inline src/utils.ts:12]: I think this will set `changed` always `true`, maybe don't improve performance in next for code

**Pull request merged**